### DISCUSSION TITLE
How can I generate a batch using several controlnet images?

### DISCUSSION THREAD

**Discussion by rpillala** [lllyasviel/stable-diffusion-webui-forge#2056] [Q&A]
I love Forge. I'm sure this is a lot of work for you and the team and I appreciate all of it. I searched for this topic but controlnet is such a popular one that I might have missed it among all the existing threads.

Maybe I don't understand the different options. In the most recent version of Controlnet I use in Automatic 1111, there is a tab that allows you to specify a folder with source images. Using this folder, A1111 will generate one image per image in the folder. That is, if there are five different images to use as openpose sources, A111 will generate five images rather than just one. If you specify a larger batch count x or batch size y, it will generate 5xy images. This does not seem to be possible using Forge. Whether I specify the image directory in "Batch Folder" or upload the images in  "Batch Upload", only the first source image is used for controlnet. I hope I'm explaining this properly.

Maybe this is going to be part of the update to integrated controlnet. I think someone described the same issue in that thread but I'm really not sure. I don't want 5 different poses to be used to  generate a single image. I would like 5 different images, one for each pose.

**Comment by AspenWiedii** [comment]
For SD 1.5 and SDXL you can accomplish the same thing by using the Img2Img tab. Drop all of the images in the batch tab on Img2Img instead of down in the ControlNet settings. The set the denoising all the way to 1. Then you can set your ControlNet options and you should get the benefit of the ControlNet from the image without [very much at all of] the influence of the image itself. This should work in Automatic1111. This has worked very well for me in Forge and Automatic1111 both. It really allows you to decide exactly how each image in your batch will affect the generated image through the ControlNet settings and the denoising together, and you can process a batch of images.
